docs/download-upload-clone/storage/high-level-design.md

```diff
@@ -0,0 +1,163 @@
+blob storage (Azure Blob Storage) for performance reasons.
+
+Whether local or remote/cloud storage is used, the backend code will use a
+single storage interface, configured to work with suitable implementations
+through instance config.
+
+## Production - cloud storage
+
+Azure Blob Storage is used to stream and store individual archive pieces.
+
+While archives are being created, pieces are stored privately, i.e. only the
+service account configured to allow the MarxanCloud backend to interact with
+Azure storage will have access to the relevant blobs.
+
+Each MarxanCloud instance makes use of a unique Azure storage account and
+related service account.
+
+Isolating tenants at the organization level may be desirable at a later stage
+for security, compliance (e.g. choice of Azure region) and billing reasons.
+
+When users request to actually download a complete archive, signed URLs are
+generated using Azure Shared Access Signature (SAS). These are nevertheless
+provided via redirection from local (API) URLs so that the eventual storage is
+transparent to the end user.
+
+## Development - local storage
+
+Local block volumes are used, transparently, as backing storage in development
+environments.
+
+More stringent size limits will apply, therefore project archive piece resolvers
+must be aware of size constraint and halt their work with a relevant error if
+available storage may not be sufficient to store a piece being prepared. This is
+so that filesystem saturation and related error states may be avoided.
+
+## Storage layout and metadata
+
+<bucketUrl>/<organizationId>/<projectId>/<exportId>
+
+The proposed URL schema is mainly meant to logically separate exports by parent
+project and organization.
+
+Blob lifecycle (for example, deleting exports at their expiration timestamp, or
+when their parent project is deleted from a MarxanCloud instance) should be
+managed through Azure blob tags.
+
+The files corresponding to pieces and final archive for each export should be
+tagged with:
+
+- `organizationId`
+- `projectId`
+- `scenarioId` (for scenario pieces)
+- `exportId`
+- `exportTimestamp` (timestamp indicating when the export was initiated;
+  identical for all pieces)
+- `expiryTimestamp` (used to programmatically remove expired exports and their
+  pieces)
+
+## Storage handling - flow
+
+This section expands with design details the summary list provided in the brief
+document.
+
+
+- storing one or more archive pieces as they are prepared and finalized
+
+For anything else than small metadata files, these pieces should be streamed to
+their eventual storage destination.
+
+- preparing a final archive file that includes all the finalized pieces
+
+This would typically be a zip (ZIP64) file. It may also be advisable to keep all
+the archive files alone and distribute them as a downloadable set of individual
+pieces instead, in order to avoid typical user challenges while handling very
+large files (e.g. > 4GiB) on personal computers.
```

Comment:
I highly doubt that anyone working in this space has their first Fat32 limitation issue using MarxanCloud. NTFS has been the default for 20+ years now. I think it's time we move on.

Reply:
I guess having worked at a IT helpdesk for some time made me irrationally over-afraid of what could go wrong on users desktops 😁 

I have no strong opinion about serving a single archive file - I used to consider this a given for data exports, but I'm not too sure anymore about its value, besides making it harder for users to accidentally delete pieces, if we distribute them alone. The 4GiB limitation is a bad example - indeed that specific legacy limitation would not be an issue in general (unless users copy stuff to a weirdly formatted usb thumb drive)